content/en/docs/reference/node/kubelet-files.md

```diff
@@ -0,0 +1,95 @@
+---
+content_type: "reference"
+title: Files written and read by the kubelet
+weight: 40
+---
+
+{{< glossary_tooltip text="Kubelet" term_id="kubelet" >}} is mostly a stateless
+process running on a Kubernetes {{< glossary_tooltip text="node" term_id="node" >}}.
+This document outlines files that kubelet reads and writes.
+
+Kubelet uses the {{< glossary_tooltip text="control plane" term_id="control-plane" >}} as
+the source of truth on what needs to run on the Node, and the
+{{<glossary_tooltip text="container runtime" term_id="container-runtime">}} to retrieve
+the current state of containers. Kubelet also relies on reading
+cgroups and various system files to collect metrics. However, there are still a few other files
+that are used by the kubelet.
+
+## Files
+
+### Configuration
+
+#### Kubelet configuration files
+
+The path to the kubelet configuration file can be configured
+using the command line argument `--config`. kubelet also supports
+[drop-in configuration files](/docs/tasks/administer-cluster/kubelet-config-file/#kubelet-conf-d)
+to enhance configuration.
+
+#### Certificates
+
+Certificates are typically located at `/var/lib/kubelet/pki`,
+but can be configured using the `--cert-dir` kubelet command line argument.
+Names of certificate files are also configurable.
+
+#### Systemd unit settings
+
+When kubelet is running as a systemd unit, some kubelet configuration may be declared
+in systemd unit settings file. Typically it includes:
+
+- command line arguments to [run kubelet](/docs/reference/command-line-tools-reference/kubelet/)
+- environment variables, used by kubelet or [configuring golang runtime](https://pkg.go.dev/runtime#hdr-Environment_Variables)
+
+### State
+
+#### Manifests
+
+Manifests for static pods are typically located in `/etc/kubernetes/manifests`.
+Location can be configured using the `staticPodPath` kubelet configuration option.
+
+#### Checkpoint files
+
+CPU managers and memory managers use state files to keep the mapping between
+Pods and allocated resources. Files are named `memory_manager_state`
+and `cpu_manager_state` and both located in the kubelet's root directory,
+typically `/var/lib/kubelet`. Kubelet root directory can be configured
+using the kubelet command line argument `--root-dir`.
+
+### Sockets
+
+Kubelet opens a few sockets as well as reading some other sockets.
+
+#### Container runtime
+
+Kubelet communicates with the container runtime using socket configured via the
+configuration parameters:
+
+- `containerRuntimeEndpoint` for runtime operations
+- `imageServiceEndpoint` for image management operations
+
+The actual values of those endpoints depend on the container runtime being used.
+
+#### Device plugins
```

Comment:
/cc @ffromani

Reply:
DRA plugins work the same way as Device Plugins, so we can probably rename this section to `Device and DRA plugins`

Reply:
added